src/components/PopupWithForm.js

```diff
@@ -5,18 +5,23 @@ export class PopupWithForm extends Popup {
     this._popUpForm = this._popUpElement.querySelector(".modal__form");
     this._handleModalSubmit = handleModalSubmit;
   }
-  closeModal() {
-    this._popUpForm.reset();
-    super.closeModal();
-  }
+  // closeModal() {
+  //   this._popUpForm.reset();
+  //   super.closeModal();
+  // }
   setEventListeners() {
     super.setEventListeners();
     this._popUpForm.addEventListener("submit", (e) => {
-      e.preventDefault();
       const modalInputs = this._getInputValues();
-      this._handleModalSubmit(modalInputs);
+      e.preventDefault();
+      this._popUpForm.querySelector(".modal__form-button").innerText =
+        "Saving...";
+      this._handleModalSubmit(modalInputs, this._popUpForm);
     });
   }
+  setSubmitAction(callBackfn) {
+    this._handleModalSubmit = callBackfn;
+  }
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`